src/razeedash/DelayedSendArray.js

```diff
@@ -32,40 +32,41 @@ const _queueObject = Symbol('queueObject');
 
 module.exports = class DelayedSendArray {
 
-  constructor(url, clusterID, max) {
+  constructor(url, clusterID, max, trackSendPromises) {
     this._clusterID = clusterID;
     this.url = url;
+    if (!this._clusterID) {
+      throw Error('clusterID must be defined');
+    }
+    if (!validUrl.isUri(url)) {
+      throw Error(`${url} not valid.`);
+    }
     if (url.startsWith('https')) {
       this.agent = httpsAgent;
     } else {
       this.agent = httpAgent;
     }
+
     this.sendObject = [];
-    if (Number.isInteger(max)) {
-      this._maxItems = max;
-    } else {
-      this._maxItems = 50;
-    }
-    if (!validUrl.isUri(url)) {
-      throw Error(`${url} not valid.`);
-    }
-    if (!this._clusterID) {
-      throw Error('clusterID must be defined');
+    this._maxItems = Number.isInteger(max) ? max : 50;
+    this._trackSendPromises = trackSendPromises || false;
+    this._sendPromises = [];
+    if (this._trackSendPromises) {
+      this._pollStarted = new Date(Date.now());
     }
   }
 
   get maxItems() {
     return this._maxItems;
   }
+  get getSendPromises() {
```

Comment:
get sendPromises() {

Reply:
i had that originally. But it looked like a "command" when you call it.. ie. `dsa.sendPromises`. so i figured it looks better as `dsa.getSendPromises`. What do you think?

Reply:
I see your pt. Consider making it getSendPromises() but I am ok either way